frontend/src/components/common/ErrorBoundary/ErrorBoundary.tsx

```diff
@@ -0,0 +1,52 @@
+import type { ComponentType, PropsWithChildren } from 'react';
+import { Component } from 'react';
+
+import { HTTPError } from '@api/HTTPError';
+
+import type { ErrorProps } from '../Error/Error';
+
+interface ErrorBoundaryProps {
+  Fallback: ComponentType<ErrorProps>;
+  onReset?: (error: Error | HTTPError) => void;
+}
+
+interface State {
+  hasError: boolean;
+  error: Error | null;
+}
+
+const initialState: State = {
+  hasError: false,
+  error: null,
+};
+
+class ErrorBoundary extends Component<PropsWithChildren<ErrorBoundaryProps>, State> {
+  state: State = initialState;
+
+  resetErrorBoundary = () => {
+    this.props.onReset?.(this.state.error!);
+    this.setState(initialState);
+  };
+
+  static getDerivedStateFromError(error: Error): State {
```

Comment:
```suggestion
 const catchError = (error: Error) => {
    setError(error);
  };
```
이런 느낌으로 구현하고 밑에서 this.props.children 대신 fallback container 만들어서 거기에 넣어주는건 어떤가요!